src/ctapipe/irf/select.py

```diff
@@ -0,0 +1,298 @@
+        }
+
+        return QTable(names=columns, units=units, descriptions=descriptions)
+
+
+class EventsLoader(Component):
+    classes = [EventPreProcessor]
+
+    def __init__(self, kind: str, file: Path, target_spectrum: Spectra, **kwargs):
+        super().__init__(**kwargs)
+
+        self.epp = EventPreProcessor(parent=self)
+        self.target_spectrum = SPECTRA[target_spectrum]
+        self.kind = kind
+        self.file = file
+
+    def load_preselected_events(
+        self, chunk_size: int, obs_time: u.Quantity
+    ) -> tuple[QTable, int, dict]:
+        opts = dict(dl2=True, simulated=True)
+        with TableLoader(self.file, parent=self, **opts) as load:
+            header = self.epp.make_empty_table()
+            sim_info, spectrum, obs_conf = self.get_metadata(load, obs_time)
+            meta = {"sim_info": sim_info, "spectrum": spectrum}
+            bits = [header]
+            n_raw_events = 0
+            for _, _, events in load.read_subarray_events_chunked(chunk_size, **opts):
+                selected = events[self.epp.get_table_mask(events)]
+                selected = self.epp.normalise_column_names(selected)
+                selected = self.make_derived_columns(selected, obs_conf)
+                bits.append(selected)
+                n_raw_events += len(events)
+
+            bits.append(header)  # Putting it last ensures the correct metadata is used
+            table = vstack(bits, join_type="exact", metadata_conflicts="silent")
+            return table, n_raw_events, meta
+
+    def get_metadata(
+        self, loader: TableLoader, obs_time: u.Quantity
+    ) -> tuple[SimulatedEventsInfo, PowerLaw, Table]:
+        obs = loader.read_observation_information()
+        sim = loader.read_simulation_configuration()
+        show = loader.read_shower_distribution()
+
+        for itm in ["spectral_index", "energy_range_min", "energy_range_max"]:
+            if len(np.unique(sim[itm])) > 1:
+                raise NotImplementedError(
+                    f"Unsupported: '{itm}' differs across simulation runs"
+                )
+
+        sim_info = SimulatedEventsInfo(
+            n_showers=show["n_entries"].sum(),
+            energy_min=sim["energy_range_min"].quantity[0],
+            energy_max=sim["energy_range_max"].quantity[0],
+            max_impact=sim["max_scatter_range"].quantity[0],
+            spectral_index=sim["spectral_index"][0],
+            viewcone_max=sim["max_viewcone_radius"].quantity[0],
+            viewcone_min=sim["min_viewcone_radius"].quantity[0],
+        )
+
+        return (
+            sim_info,
+            PowerLaw.from_simulation(sim_info, obstime=obs_time),
+            obs,
+        )
+
+    def make_derived_columns(self, events: QTable, obs_conf: Table) -> QTable:
+        if obs_conf["subarray_pointing_lat"].std() < 1e-3:
+            assert all(obs_conf["subarray_pointing_frame"] == 0)
+            # Lets suppose 0 means ALTAZ
```

Comment:
You don't need  to suppose: 


https://github.com/cta-observatory/ctapipe/blob/80914fb888e09cf1bf145c330c813a5be3e6efda/src/ctapipe/containers.py#L134-L143